src/e3/python/pypi.py

```diff
@@ -357,9 +357,10 @@ def add_constraint(self, requirement: Requirement) -> None:
             self.versions = [
                 v
                 for v in self.versions
-                if packaging.specifiers.Specifier(f"{spec[0]}{spec[1]}").contains(
-                    str(v)
-                )
+                if packaging.specifiers.Specifier(
```

Comment:
Knowing `packaging.specifiers` is used only here, we should rather import through `from packaging.specifiers import Specifier`, and use `Specifier` here.

Reply:
I agree, I will make the change.

Reply:
Done.